modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -0,0 +1,968 @@
+	var/obj/item/organ/genital/testicles/penis = owner.getorganslot(ORGAN_SLOT_PENIS)
+
+	if(H.client?.prefs.erp_pref == "Yes")
+		if(penis && balls && H.wear_condom())
+			if(prob(40))
+				H.emote("moan")
+			balls.reagents.remove_all(balls.reagents.total_volume * 0.6)
+			var/obj/item/clothing/sextoy/condom/C = H.get_item_by_slot(ITEM_SLOT_PENIS)
+			C.condom_use()
+			if(C.condom_state == "broken")
+				var/turf/T = get_turf(H)
+				new /obj/effect/decal/cleanable/cum(T)
+
+		if(balls && H.is_bottomless() && !H.wear_condom())
+			var/turf/T = get_turf(H)
+			new /obj/effect/decal/cleanable/cum(T)
+			if(prob(40))
+				H.emote("moan")
+			balls.reagents.remove_all(balls.reagents.total_volume * 0.6)
+
+		if(vagina && H.is_bottomless()) //Sry, futanari players, but condom's don't work like that for vagina, so yep.
+			var/turf/T = get_turf(H)
+			new /obj/effect/decal/cleanable/femcum(T)
+			if(prob(40))
+				H.emote("moan")
+			vagina.reagents.remove_all()
+
+	return ..()
+
+////////////////////////
+///SPANKING PROCEDURE///
+////////////////////////
+
+//Hips are red after spanking
+/datum/status_effect/spanked
+	id = "spanked"
+	duration = 300 SECONDS
+	alert_type = null
+
+/mob/living/carbon/human/examine(mob/user)
+	.=..()
+	var/t_his = p_their()
+	var/mob/living/U = user
+
+	if(stat != DEAD && !HAS_TRAIT(src, TRAIT_FAKEDEATH) && src != U)
+		if(src != user)
+			if(has_status_effect(/datum/status_effect/spanked) && is_bottomless())
+				. += "<font color=purple>[t_his] thighs turned red.</font>\n"
+
+//Mood boost for masochist
+/datum/mood_event/perv_spanked
+	description = "<font color=purple>Ah, yes! More! Punish me!</font>\n"
+	mood_change = 3
+	timeout = 5 MINUTES
+
+/////////////////////
+///SUBSPACE EFFECT///
+/////////////////////
+
+/datum/status_effect/subspace
+	id = "subspace"
+	tick_interval = 10
+	duration = 5 MINUTES
+	alert_type = null
+
+/datum/status_effect/subspace/on_apply()
+	. = ..()
+	var/mob/living/carbon/human/M = owner
+	if(owner.masochism == FALSE)
+		owner.set_masochism(TRUE)
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "subspace", /datum/mood_event/subspace)
+
+/datum/status_effect/subspace/on_remove()
+	. = ..()
+	var/mob/living/carbon/human/M = owner
+	if(owner.masochism == TRUE && !HAS_TRAIT(owner, TRAIT_MASOCHISM))
+		owner.set_masochism(FALSE)
+	SEND_SIGNAL(M, COMSIG_CLEAR_MOOD_EVENT, "subspace", /datum/mood_event/subspace)
+
+/datum/mood_event/subspace
+	description = "<font color=purple>Everything so woozy... Pain feels so... Awesome.</font>\n"
+	mood_change = 4
+
+///////////////////////
+///AROUSAL INDICATOR///
+///////////////////////
+
+/obj/item/organ/brain/on_life(delta_time, times_fired) //All your horny is here *points to the head*
+	. = ..()
+	var/mob/living/carbon/human/H = owner
+	if(istype(owner, /mob/living/carbon/human) && H.client?.prefs.erp_pref == "Yes")
+		if(!(organ_flags & ORGAN_FAILING))
+			H.dna.species.handle_arousal(H, delta_time, times_fired)
```

Comment:
I'll be honest, it's just going to continue like this forever for the rest of this file. You should understand by now what you have to do. Get rid of all those 1-letter variables. Coding standards.
```suggestion
	var/mob/living/carbon/human/brain_owner = owner
	if(istype(brain_owner, /mob/living/carbon/human) && brain_owner.client?.prefs.erp_pref == "Yes")
		if(!(organ_flags & ORGAN_FAILING))
			brain_owner.dna.species.handle_arousal(H, delta_time, times_fired)
```